### PR TITLE
[bsp][pico]update: add drv_soft_spi

### PR DESCRIPTION
**Pull request by Z8MAN8**
## 拉取/合并请求描述：(PR description)

[

#### 在什么测试环境下测试通过 (what is the test environment)

]

### 当前拉取/合并请求的状态 Intent for your PR

必须选择一项 Choose one (Mandatory):

- [ ] 本拉取/合并请求是一个草稿版本 This PR is for a code-review and is intended to get feedback
- [ ] 本拉取/合并请求是一个成熟版本 This PR is mature, and ready to be integrated into the repo

### 代码质量 Code Quality：

我在这个拉取/合并请求中已经考虑了 As part of this pull request, I've considered the following:

- [ ] 已经仔细查看过代码改动的对比 Already check the difference between PR and old code
- [ ] 代码风格正确，包括缩进空格，命名及其他风格 Style guide is adhered to, including spacing, naming and other styles
- [ ] 没有垃圾代码，代码尽量精简，不包含`#if 0`代码，不包含已经被注释了的代码 All redundant code is removed and cleaned up
- [ ] 所有变更均有原因及合理的，并且不会影响到其他软件组件代码或BSP All modifications are justified and not affect other components or BSP
- [ ] 对难懂代码均提供对应的注释 I've commented appropriately where code is tricky
- [ ] 代码是高质量的 Code in this PR is of high quality
- [ ] 已经使用[formatting](https://github.com/mysterywolf/formatting)等源码格式化工具确保格式符合[RT-Thread代码规范](https://github.com/RT-Thread/rt-thread/blob/master/documentation/contribution_guide/coding_style_cn.md) This PR complies with [RT-Thread code specification](https://github.com/RT-Thread/rt-thread/blob/master/documentation/contribution_guide/coding_style_en.md)

### CODE REVIEW

File: bsp/raspberry-pico/drivers/SConscript [hunk 1]
```diff
@@ -1,8 +1,16 @@
 from building import *
 
 cwd = GetCurrentDir()
-src = Glob('*.c')
+src = []
 CPPPATH = [cwd]
+src += ['drv_gpio.c']
+src += ['drv_uart.c']
```
Comment (mysterywolf) [inline bsp/raspberry-pico/drivers/SConscript:7]: 这两个也得用宏控制一下，只不过一般来说 这两个在Kconfig里默认是y

**Review by mysterywolf — APPROVED** [review]

File: bsp/raspberry-pico/drivers/drv_soft_spi.h [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+ * Change Logs:
+ * Date           Author       Notes
+ * 2023-06-02     Chushicheng  the first version
+ */
+#ifndef DRV_SOFT_SPI_H_
+#define DRV_SOFT_SPI_H_
+
+#ifdef BSP_USING_SOFT_SPI
+#include <rtdevice.h>
+#include <spi-bit-ops.h>
```
Comment (mysterywolf) [inline bsp/raspberry-pico/drivers/drv_soft_spi.h:15]: 看一下这个头文件可否能收缩到.c文件里？

**Comment by supperthomas** [comment]
下次可以尝试用driver中的soft spi对接试试~

**Review by supperthomas — APPROVED** [review]

**Comment by supperthomas** [comment]
https://github.com/RT-Thread/rt-thread/blob/master/components/drivers/spi/spi-bit-ops.c

**Review by Guozhanxin — APPROVED** [review]

**Comment by BernardXiong** [comment]
又是一个ci报错就直接合并的，这样ci就形同虚设了。。。

**Pull request merged**